api/utils/db.py

```diff
@@ -103,6 +103,35 @@ async def dependable_get_write_project_connection(
     )
 
 
+async def HACK_dependable_contributor_project_connection(
+    project: str,
+    author: str = Depends(authenticate),
+    ar_guid: str = Depends(get_ar_guid),
+    extra_values: dict | None = Depends(get_extra_audit_log_values),
+) -> Connection:
+    """FastAPI handler for getting connection WITH project"""
+    meta = {}
+    if extra_values:
+        meta.update(extra_values)
+
+    meta = {'role': 'contributor'}
```

Comment:
Probably wants to go above the `meta.update(extra_values)` bit?

Reply:
I didn't want any other values to potentially update it?

Reply:
Fair, but then it would be better as

```
meta['role'] ='contributor'
```

At the moment, it wipes out anything passed in as an argument…

Reply:
Oh wow, I'm losing my marbles two ways 😓